src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -656,75 +896,288 @@ private static unsafe nuint GetIndexOfFirstNonAsciiChar_Default(char* pBuffer, n
 
             // Before we drain off char-by-char, try a generic vectorized loop.
             // Only run the loop if we have at least two vectors we can pull out.
-
-            if (Vector.IsHardwareAccelerated && bufferLength >= 2 * (uint)Vector<ushort>.Count)
+            if (Vector512.IsHardwareAccelerated && bufferLength >= 2 * (uint)Vector512<ushort>.Count)
             {
-                uint SizeOfVectorInChars = (uint)Vector<ushort>.Count; // JIT will make this a const
-                uint SizeOfVectorInBytes = (uint)Vector<byte>.Count; // JIT will make this a const
+                uint SizeOfVector512InChars = (uint)Vector512<ushort>.Count; // JIT will make this a const
+                uint SizeOfVector512InBytes = (uint)Vector512.Size; // JIT will make this a const
```

Comment:
You should be able to use `Vecto512.Size` directly and have C# keep it a const instead.

It will make things a bit more readable and allow C# to constant fold some of the simple cases (e.g. `~(nuint)(SizeOfVector512InBytes - 1)` itself will become constant foldable rather than forcing IL to emit `ldloc; sub; conv.i; sub` and then requiring the JIT to optimize that down.